child_compassion/models/child_pictures.py

```diff
@@ -88,67 +90,73 @@ def create(self, vals):
         if existing_picture:
             return existing_picture
 
-        pictures = super().create(vals)
-        # Retrieve Headshot
-        image_date = pictures._get_picture("Headshot", width=180, height=180)
-        # Retrieve Fullshot
-        image_date = image_date and pictures._get_picture(
-            "Fullshot", width=800, height=1200
-        )
+        picture = super().create(vals)
+        # Fetch the image from the webservice
+        image_date = self._fetch_and_attach_pictures(picture)
+
+        # Handle the case where the picture cannot be fetched
         if not image_date:
-            # We could not retrieve a picture, we cancel the creation
-            pictures.child_id.message_post(
-                body=_(pictures._error_msg), subject=_("Picture update")
-            )
-            pictures.unlink()
-            return False
+            self._handle_picture_fetch_failure(picture)
```

Comment:
No need to pass a record as a parameter of the function, as it can be passed via the `self` parameter when calling the method directly from the record like that:
```suggestion
            picture._handle_picture_fetch_failure()
```